nautobot/extras/forms.py

```diff
@@ -270,6 +274,179 @@ def save(self, commit=True):
+            else:
+                side = initial_side
+            peer_side = RelationshipSideChoices.OPPOSITE[side]
+
+            # If this model is on the "source" side of the relationship, then the field will be named
+            # "cr_<relationship-slug>__destination" since it's used to pick the destination object(s).
+            # If we're on the "destination" side, the field will be "cr_<relationship-slug>__source".
+            # For a symmetric relationship, both sides are "peer", so the field will be "cr_<relationship-slug>__peer"
+            field_name = f"cr_{relationship.slug}__{peer_side}"
+
+            if field_name in self.relationships:
+                # This is a symmetric relationship that we already processed from the opposing "initial_side".
+                # No need to process it a second time!
+                continue
+
+            if relationship.has_many(side):
+                if relationship.type == RelationshipTypeChoices.TYPE_ONE_TO_MANY:
+                    # Destination side of a one-to-many field - provide a standard form field for selecting the "one",
+                    # as well as making this field nullable.
+                    self.fields[field_name] = relationship.to_form_field(side=side)
+                    self.nullable_fields.append(field_name)
+                else:
+                    # Many-to-many field - provide "add" and "remove" form fields like with tags, no nullable option.
+                    self.fields[f"add_{field_name}"] = relationship.to_form_field(side=side)
+                    self.fields[f"add_{field_name}"].label = "Add " + self.fields[f"add_{field_name}"].label
+                    self.fields[f"remove_{field_name}"] = relationship.to_form_field(side=side)
+                    self.fields[f"remove_{field_name}"].label = "Remove " + self.fields[f"remove_{field_name}"].label
+            else:
+                # The "one" side of a one-to-one or one-to-many relationship.
+                # In this case, the only valid bulk-edit operation is nulling/clearing the relationship,
+                # but the "Set null" checkbox only appears if we have a form field for the the relationship itself.
+                # This could probably be refined, but for now we just add the field and disable it.
+                self.fields[field_name] = relationship.to_form_field(side=side)
+                self.fields[field_name].disabled = True
+                self.nullable_fields.append(field_name)
+
+            self.relationships.append(field_name)
+
+    def save_relationships(self, *, instance, nullified_fields):
+        """Helper method to be called from BulkEditView.post()."""
+        # TODO the below is inefficient as it re-loads the Relationship objects afresh for each instance;
+        # however this is necessary at present as applies the source/destination filters (if any) to determine
+        # whether each relationship actually applies to the given instance.
```

Comment:
I get why this feels inefficient but if there isn't a clear way to make it efficient I think it's worth removing the `TODO`. This could be the best we are going to get and unless we receive a clear performance penalty on submitting a bulk edit (much faster than doing some of these operations manually...) it's pretty good as it stands.

Reply:
```suggestion
        # The below may seem inefficient as it re-loads the Relationship objects afresh for each instance;
        # however this is necessary as it applies the source/destination filters (if any) to determine
        # whether each relationship actually applies to the given instance.
```